graphql_compiler/compiler/ir_lowering_sql/metadata.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2018-present Kensho Technologies, LLC.
+from ... import exceptions
+from .constants import SqlBackend
+
+
+class SqlMetadata(object):
+    """Metadata wrapper for use during compilation.
+
+    In order to transform GraphQL to SQL, there needs to be additional configuration specified
+    for mapping:
+        - GraphQL types -> SQL tables
+        - GraphQL fields -> SQL columns
+        - GraphQL Edges -> SQL JOINs
+    """
+
+    def __init__(self, dialect, sqlalchemy_metadata):
+        """Initialize a new sql metadata manager."""
+        self.sqlalchemy_metadata = sqlalchemy_metadata
+        self._db_backend = SqlBackend(dialect)
+        self.table_name_to_table = {
+            name.lower(): table for name, table in self.sqlalchemy_metadata.tables.items()
```

Comment:
`for` and onward on the next line please, this is pretty long. It might also be a good idea to use `six.iteritems()` instead of `.items()` here.